app/models/result.rb

```diff
@@ -148,6 +148,11 @@ def check_for_nil_marks(user_visibility = :ta)
     # want the peer-visible criteria
     visibility = is_a_review? ? :peer : user_visibility
 
+    if submission == nil
```

Comment:
This is the wrong place for this check. You should be able to do it before creating the result.